jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/ObtainBaseImageLayerStep.java

```diff
@@ -131,15 +133,20 @@ private ObtainBaseImageLayerStep(
 
   @Override
   public PreparedLayer call() throws IOException, CacheCorruptedException, RegistryException {
+    EventHandlers eventHandlers = buildConfiguration.getEventHandlers();
     DescriptorDigest layerDigest = layer.getBlobDescriptor().getDigest();
     try (ProgressEventDispatcher progressEventDispatcher =
             progressEventDispatcherFactory.create("checking base image layer " + layerDigest, 1);
         TimerEventDispatcher ignored =
-            new TimerEventDispatcher(
-                buildConfiguration.getEventHandlers(), String.format(DESCRIPTION, layerDigest))) {
+            new TimerEventDispatcher(eventHandlers, String.format(DESCRIPTION, layerDigest))) {
 
       StateInTarget stateInTarget = blobExistenceChecker.check(layerDigest);
       if (stateInTarget == StateInTarget.EXISTING) {
+        eventHandlers.dispatch(
+            LogEvent.info(
+                "Pull skipped for BLOB : "
```

Comment:
why would we skip a *pull* if image is already in the registry?

Reply:
should it be `" already exists in local cache"`

Reply:
If the base image layer exists on the target already, we don't pull the layer.

Reply:
Can we rearrange the words, because `BlobDescriptor.toString()` returns `disgest: <...>, size: <size>`, so the sentence may feel disconnected. And the digest string is long, so I think it's better to print it last.